ACM-18043: Policy templating doc tweaks (GRC) (#7525)
open-cluster-management/rhacm-docs: governance/adv_template_process.adoc

Continue reading for advanced template use-cases:

* <<special-annotation-processing,Special annotation for reprocessing>>

> * <<bypass-template-processing,Bypass template processing>>

Reviewer: Title of the section reads "Bypassing template processing"

> * <<bypass-template-processing,Bypass template processing>>

Reviewer: ```suggestion
* <<bypass-template-processing,Bypassing template processing>>
```

> * <<raw-object-template-processing,Raw object template processing>>

Reviewer: Title of the section reads "Processing raw templates in configuration policies"

Additionally, I wonder if you might want to drop the "configuration" like "Processing raw templates in policies"? The level 1 header was changed from "configuration policies" to just "policies".

> * <<raw-object-template-processing,Raw object template processing>>

Reviewer: ```suggestion
* <<raw-object-template-processing,Processing raw templates in configuration policies>>
```

If you need to manually initiate an update, use the special annotation, `policy.open-cluster-management.io/trigger-update`, to indicate changes for the data referenced by the templates. Any change to the special annotation value automatically initiates template processing. Additionally, the latest contents of the referenced resource are read and updated in the policy definition that is propagated for processing on managed clusters. A way to use this annotation is to increment the value by one each time.

[#bypass-template-processing]
== Bypassing template processing

> By default, {acm-short} processes all Go templates. You might create a policy that contains a policy in `policy-templates` that is not intended to be processed by {acm-short}. To bypass template processing for a particular template, change `{{ template content }}` to `+{{ `{{ template content }}` }}+` to have the template return a raw string to be processed by a subsequent templating engine.

Reviewer: It might make sense to divide the sentence into two. Something like:

```suggestion
By default, {acm-short} processes all Go templates. You might create a policy that contains a policy in `policy-templates` that is not intended to be processed by {acm-short}. To bypass template processing for a particular template, change `{{ template content }}` to `+{{ `{{ template content }}` }}+`. Then, the template returns a raw string to be processed by a subsequent templating engine.
```

> Alternatively to bypass template resolution entirely for a configuration policy, add the following annotation in the relevant `ConfigurationPolicy` contained in your `Policy`: `policy.open-cluster-management.io/disable-templates: "true"`.

Reviewer 2: Instead of presenting this as an alternative option, what do you think about presenting is a separate use case?

```suggestion
If you want to completely bypass the template resolution for a configuration policy, add the following annotation in the relevant `ConfigurationPolicy` contained in your `Policy`: `policy.open-cluster-management.io/disable-templates: "true"`.
```

Author: Yes, that sounds good to me!

> If you want to completely bypass the template resolution for a configuration policy, add the following annotation in the relevant `ConfigurationPolicy` contained in your `Policy`: `policy.open-cluster-management.io/disable-templates: "true"`.

Reviewer: I'm not sure what is the connection between the `ConfigurationPolicy` and the `Policy`  and what type of resource they are exactly but I'd rephrase this a bit:

```suggestion
If you want to completely bypass the template resolution for a configuration policy, add the `policy.open-cluster-management.io/disable-templates: "true"` annotation in the relevant `ConfigurationPolicy` that is contained in your `Policy` resource.
```

[#raw-object-template-processing]
== Processing raw templates in configuration policies

The `object-template-raw` parameter is an optional parameter that supports advanced templating use-cases, such as the `if` conditional function and the `range` loop function. The `object-templates-raw` parameter accepts a string containing Go templates, and when you use these templates, they must result in an `object-templates` array.

> For example, see the following YAML sample that adds the `species-category: mammal` label to any ConfigMap in the `default` namespace that has a `name` key equal to `Sea Otter`:

Reviewer: ```suggestion
For example, see the following YAML sample that adds the `species-category: mammal` label to any `ConfigMap` in the `default` namespace that has a `name` key equal to `Sea Otter`:
```
